### PR TITLE
Adapt TuYa converters to multiple data point change

### PR DESCRIPTION
**Pull request by gmbnomis**
Receiving side:

Types `commandDataReport`, `commandDataResponse`, and
`commandActiveStatusReport` receive an array of DP values now (instead of a
single value).  Adapt existing converters to use the first entry of this
array.
    
To avoid breakage, additional DP values that may be present in a payload are
ignored (as before).

Additionally, extend `tuya_data_point_dump` to support multiple DP values per payload
    
Adapt `scripts/read_tuya_dump.py` accordingly and make its output more
helpful (print the numerical DP value and remove the `fn` field).  Its
output for a packet with two DP values now looks like this:
    
```
2021-12-22T20:50:13 0x867829fffe2de210 (seq:    0, value: #0) [DP   1 unknown     ] => enum: 1
2021-12-22T20:50:13 0x867829fffe2de210 (seq:    0, value: #1) [DP   4 unknown     ] => int: 92
```

Sending side:

The existing helper functions can still be used to send a single DP value. Add a new function `sendDataPoints` that
is able to send multiple DP values and helpers `dpValueFrom*` to construct DP values.

### CODE REVIEW

**Comment by gmbnomis** [comment]
This PR requires https://github.com/Koenkk/zigbee-herdsman/pull/483 to be merged first

**Review by hacker-cb — CHANGES_REQUESTED** [review]

File: devices/zemismart.js [hunk 1]
```diff
@@ -11,8 +11,9 @@ const fzLocal = {
         cluster: 'manuSpecificTuya',
         type: ['commandGetData', 'commandSetDataResponse'],
         convert: (model, msg, publish, options, meta) => {
-            const button = msg.data.dp;
-            const actionValue = tuya.getDataValue(msg.data.datatype, msg.data.data);
+            const dpValue = msg.data.dpValues[0];
```
[the remark this answers is not in the document]
Comment (gmbnomis) [reply devices/zemismart.js:14]: This is definitely how new converters should be written. See [an example](https://github.com/Koenkk/zigbee2mqtt.io/blob/a675a0d53a77112aac8b43786b8e4237f24f06e6/docs/advanced/support-new-devices/02_support_new_tuya_devices.md#5-adding-your-first-data-point) in my proposed change to the "Support new Tuya devices" page.

However, for existing converters, this means to introduce larger (structural) changes:

1. Most converters return on the first matched value, i.e. a for loop would only make a difference if data would be gathered for all DP values and returned after the loop (as in the documentation example).
2. I have no idea what happens if we begin to look at additional DP values.

For 1., I am afraid to introduce breakage, as we don't have tests for converters. For 2. I am afraid to introduce breakage as I don't have any of those device. 

Additionally, we don't know if these changes would make any difference (if a device never sends multiple DP values, the best we can do is not to break things). And it is a lot of additional work, the current replacements are mostly "robotic".

To play safe here, I did not introduce loops in the converters. 

What we could do is to introduce warning logs if there are additional DP values indeed. We could have a `firstDpValue(msg, meta, converterName)` function. As an example:

```js
    tuya_water_leak: {
        cluster: 'manuSpecificTuya',
        type: 'commandDataReport',
        convert: (model, msg, publish, options, meta) => {
            const dpValue = tuya.firstDpValue(msg, meta, 'tuya_water_leak');
   ...
```

This could log a warning that DP values were ignored and suggest to use a for loop including a pointer to the "Support new Tuya devices" page". However, the call is somewhat awkward in order to pass the information needed for logging.

In conclusion, I see the following options:

1. Bite the bullet and convert everything to loops (but perhaps in smaller chunks, i.e. merge this PR and begin to change to loops in subsequent PRs)
2. Keep it as in the current PR
3. Introduce logs (which would only emit warnings if multiple DPs occur in practice)
4. ?

I think I prefer option 3.

@hacker-cb What do you think?
@Koenkk How are such migrations handled in the zigbee2mqtt project typically?

[the remark this answers is not in the document]
Comment (Koenkk) [reply devices/zemismart.js:14]: 3 seems the best indeed, usually I merge such pr directly after release to have the longest possible test exposure (a new z2m release is made every 1st of the month)

**Comment by hacker-cb** [comment]
I agree, option 3 is better.
Also it will be nice to send anonymous reports about such situations to the statistics server.

File: devices/zemismart.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (hacker-cb) [reply devices/zemismart.js:14]: @hacker-cb hacker-cb commented now
I agree, option 3 is better.
Also it will be nice to send anonymous reports about such situations to the statistics server.

[the remark this answers is not in the document]
Comment (Koenkk) [reply devices/zemismart.js:14]: > Also it will be nice to send anonymous reports about such situations to the statistics server.

There is no statistics server (and I prefer to keep it such for privacy reasons). Better to log with a proper message on what to do

[the remark this answers is not in the document]
Comment (gmbnomis) [reply devices/zemismart.js]: I adapted the PR accordingly, log message looks like this:

`zigbee-herdsman-converters:tuya_water_leak: Additional DP #4 with data {"dp":4,"datatype":2,"data":[0,0,0,75]} will be ignored! Use a for loop in the fromZigbee converter (see https://www.zigbee2mqtt.io/advanced/support-new-devices/02_support_new_tuya_devices.html)`

[the remark this answers is not in the document]
Comment (Koenkk) [reply devices/zemismart.js]: Looks good to me!
